### ISSUE TITLE
make use of on_missing_core_dims in apply_ufunc

### ISSUE THREAD

**Issue by mathause** [MESMER-group/mesmer#293]
/pydata/xarray#8138 added an option to ignore missing core dims in `apply_ufunc`. We currently use our home baked version for this in `lowess`. I thought there are more cases but these don't use `apply_ufunc`.

I think this is a first good issue of medium complexity (because a version check is needed) but low priority ;-)

https://github.com/MESMER-group/mesmer/blob/7c5fdba2fd23ee870f81dd9e13d99f5fe4ee5183/mesmer/stats/smoothing.py#L79

**Comment by veni-vidi-vici-dormivi** [comment]
Can you elaborate on the version check? Do you mean we need to check if xarray > v2023.09.0 and if not apply the home-baked version?

**Comment by mathause** [comment]
> Do you mean we need to check if xarray > v2023.09.0 and if not apply the home-baked version?

Yes. However, I am currently doubting it's worth the hassle (I'll leave it up to you). For an example how to do version checks:

https://github.com/MESMER-group/mesmer/blob/cb21b503b95eaf92e326e64aace7e845ade7c79a/tests/unit/test_linear_regression.py#L229
